docker-app/qfieldcloud/core/admin.py

```diff
@@ -982,29 +1016,54 @@ def set_status_unpermitted(self, request, queryset):
 
     def response_change(self, request, delta):
         if "_apply_delta_btn" in request.POST:
-            if not delta.project.project_filename:
-                self.message_user(request, "Missing project file")
-                raise exceptions.NoQGISProjectError()
-
-            if not jobs.apply_deltas(
-                delta.project,
-                request.user,
-                delta.project.project_filename,
-                delta.project.overwrite_conflicts,
-                delta_ids=[str(delta.id)],
-            ):
-                self.message_user(request, "No deltas to apply")
-                raise exceptions.NoDeltasToApplyError()
-
-            self.message_user(request, "Delta application started")
-
-            # we need to sleep 1 second, just to make sure the apply delta started
-            time.sleep(1)
-
-            return HttpResponseRedirect(".")
+            return self.apply_delta(request, delta)
 
         return super().response_change(request, delta)
 
+    def apply_delta(self, request, delta):
+        if not delta.project.project_filename:
+            self.message_user(request, "Missing project file")
+            raise exceptions.NoQGISProjectError()
+
+        if not jobs.apply_deltas(
+            delta.project,
+            request.user,
+            delta.project.project_filename,
+            delta.project.overwrite_conflicts,
+            delta_ids=[str(delta.id)],
+        ):
+            self.message_user(request, "No deltas to apply")
+            raise exceptions.NoDeltasToApplyError()
+
+        self.message_user(request, "Delta application started")
+
+        # we need to sleep 1 second, just to make sure the apply delta started
+        time.sleep(1)
+
+        return HttpResponseRedirect(".")
+
+    @admin.action(description="Download deltas as deltafile")
+    def download_deltafile(self, request, queryset: QuerySet) -> HttpResponse:
+        if queryset.values_list("project_id").distinct().count() != 1:
+            messages.error(request, "Selected deltas must be part of the same project!")
```

Comment:
It would be nice to have this information upfront, like in the action title `Download project deltas as file`. But I guess is hard to provide in a concise way. Enabling cross project downloads would be undesired or hard to implement?

Reply:
Cross project downloads are undesired. Keep in mind this is more for us rather than support or anyone else. So no need to be extra explanatory.